backend/permissions.py

```diff
@@ -1,18 +1,25 @@
 from rest_framework import permissions
 
-class IsAdminOrSupplier(permissions.BasePermission):
+class IsSupplier(permissions.BasePermission):
     """
-    Разрешение, которое позволяет доступ только администраторам или пользователям с правами is_supplier.
+    Разрешение, позволяющее доступ только поставщикам.
+    """
+    
+    def has_permission(self, request, view):
+        return request.user.is_authenticated and request.user.role == 'supplier'
```

Comment:
Тогда получится примерно так и ты избавишься от двух ненужных классов
```suggestion
    def has_permission(self, request, view, role):
        return request.user.is_authenticated and request.user.role == role
```

Reply:
OK

Reply:
Сделано